File: app/src/main/java/com/blockeq/stellarwallet/remote/SupportedAssetsApi.java
package com.blockeq.stellarwallet.remote;

import com.blockeq.stellarwallet.models.SupportedAsset;

import java.util.Map;

import retrofit2.Call;
import retrofit2.http.GET;

public interface SupportedAssetsApi {
    @GET("directory/assets")
    Call<Map<String, SupportedAsset>> getAssetsDirectory();
}

File: app/src/main/java/com/blockeq/stellarwallet/remote/SupportedAssetsApi.kt
package com.blockeq.stellarwallet.remote

import com.blockeq.stellarwallet.models.SupportedAsset

import retrofit2.Call
import retrofit2.http.GET

interface SupportedAssetsApi {
    @get:GET("/directory/assets")
    val assets: Call<Map<String, SupportedAsset>>
}
